docs/BEMAN_LIBRAY_MATURITY_MODEL.md

```diff
@@ -0,0 +1,35 @@
+<!--
+SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+-->
+
+# The Beman Library Maturity Model
+
+This document specifies the Beman Library Maturity Model.
+Its goal is to create awareness about the current status of Beman libraries.
+
+## Introduction
+
+### Changing this document
+
+<!-- TODO: Maybe instead of duplicatind content from BEMAN_STANDARD.md#Changing-this-document here, we put a single section in root README.md -->
+
+## Beman library maturity model
+
+The Beman maturity model helps developers quickly assess the production readiness of Beman libraries by classifying them based on development phase and interface stability.
+
+<img src="./img/beman-library-maturity-model.png">
+
+### Under development and not yet ready for production use.
+<img src="./img/logo-beman-library-under-development.png" style="width:5%; height:auto;"> These libraries may deviate from the Beman Standard due to incompleteness, lack of testing, inconsistencies with the specification, or other non-conformances. They are not recommended for production usage.
+
+### Production ready. API may undergo changes.
+<img src="./img/logo-beman-library-production-ready-api-may-undergo-changes.png" style="width:5%; height:auto;"> These Beman-compliant libraries are production-ready, fully implementing the target paper with complete testing and documentation. Users should be aware that future API changes are possible and that standardization is not guaranteed.
+
+### Production ready. Stable API.
+<img src="./img/logo-beman-library-production-ready-stable-api.png" style="width:5%; height:auto;"> These production-ready libraries offer stable, standardized APIs.  They are part of the C++ Standard and can be used as a polyfill for compilers lacking native support. Note that these libraries will be retired after two standardization cycles (6 years).
+
+### Retired. No longer maintained or actively developed.
+<img src="./img/logo-beman-library-retired.png" style="width:5%; height:auto;"> These libraries are not recommended for production use. They were completed and [Production ready. Stable API.](./BEMAN_LIBRAY_MATURITY_MODEL.md#production-ready-stable-api) at some point, but they are no longer developed or maintained, because they have been superseded by native compiler implementations.
+
+### Dropped. Never got completed.
+<img src="./img/logo-beman-library-dropped.png" style="width:5%; height:auto;"> These libraries should not be used for production. They were not got completed because they were not accepted into the the C++ Standard.
```

Comment:
My understanding is that we may want to distinguish between a library which was production ready at some point and now it's just retired (it may still be good enough for an old codebase) vs a library which was never ready to use.

Last one represents something unusable.

First one represents something which may be still usable for some users.

Reply:
> Last one represents something unusable.

Both could be usable. Consider a library that meets all the Beman specifications whose proposal landed in the working draft and got pulled at the last minute.

Reply:
But that doesn't cover the case where someone made an excellent proposal with an excellent implementation that matched it and it got rejected for some reason. That kind of thing is also very usable. I think we're making this too complicated.

Reply:
Would changing `Dropped. Never completed.` to `Dropped.`?

New description: "Most of these libraries are not suitable for production use because they were not accepted into the C++ Standard and, as a result, were never fully completed. There may be exceptions, such as certain implementations that were production-ready at some point but were ultimately rejected during the final stages of the ISO C++ Standardization process. If you are particularly interested in using one of these libraries, please reach out to the Beman community on discourse."

Reply:
I think `Dropped` is good enough, also archiving the repository to suggest the discontinuing of the work.

Reply:
I will close this thread as per https://github.com/bemanproject/beman/pull/72#issuecomment-2552050751.